### PR TITLE
NTP SI V2

### PR DESCRIPTION
**Pull request by simonhong**
Resolves https://github.com/brave/brave-browser/issues/12190

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [ ] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java [hunk 1]
```diff
@@ -520,7 +520,7 @@ public void logoRetrieved(Wallpaper mWallpaper, Bitmap logoWallpaper) {
                     @Override
                     public void onClick(View view) {
                         if (mWallpaper.getLogoDestinationUrl() != null) {
-                            NTPUtil.openUrlInSameTab(mWallpaper.getLogoDestinationUrl());
+                            mNTPBackgroundImagesBridge.wallpaperLogoClicked(mWallpaper);
```
Comment (simonhong) [inline android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:523]: @tmancey With this change, all logo click event will be delegated to ads service.
Is this ok? Or only should be delegated if current wallpaper has proper id strings?

Comment (tmancey) [reply android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:523]: This is perfect, as ads can then validate and open the url

File: components/brave_new_tab_ui/components/default/brandedWallpaper/logo.tsx [hunk 2]
```diff
@@ -11,13 +11,14 @@ import * as Styled from './logo-style'
...
+  onClickLogo: () => void
 }
 
-function Logo ({ data }: Props) {
+function Logo ({ data, onClickLogo }: Props) {
   return (
     <>
       <Styled.Image src={data.image} alt={data.alt} />
-      <Styled.Anchor href={data.destinationUrl} title={data.alt}>
+      <Styled.Anchor title={data.alt} onClick={onClickLogo}>
```
Comment (simonhong) [inline components/brave_new_tab_ui/components/default/brandedWallpaper/logo.tsx:21]: @petemill Can you check this change? With this, link click is delegated to browser.

File: components/ntp_background_images/browser/view_counter_service.cc [hunk 3]
```diff
@@ -68,6 +69,18 @@ ViewCounterService::ViewCounterService(NTPBackgroundImagesService* service,
...
 
+void ViewCounterService::BrandedWallpaperWillBeDisplayed(
+    const std::string& wallpaper_id) {
+  base::Value data = ViewCounterService::GetCurrentWallpaperForDisplay();
+  DCHECK(!data.is_none());
+  const std::string id = *data.FindStringKey(kCreativeInstanceIDKey);
+  LOG(ERROR) << "CreativeInstanceID: " << id;
+  LOG(ERROR) << "WallpaperID: " << wallpaper_id;
+  if (!id.empty()) {
+    // TODO(Terry): Ping NTP displayed event to ads service.
```
Comment (simonhong) [inline components/ntp_background_images/browser/view_counter_service.cc:80]: @tmancey This is where we should ping about NTP SI displaying

File: components/ntp_background_images/browser/view_counter_service.cc [hunk 4]
```diff
@@ -178,6 +191,17 @@ void ViewCounterService::RegisterPageView() {
...
 
+void ViewCounterService::BrandedWallpaperLogoClicked(
+    const std::string& creative_instance_id,
+    const std::string& destination_url,
+    const std::string& wallpaper_id) {
+  LOG(ERROR) << ": creative instance id: " << creative_instance_id;
+  LOG(ERROR) << ": destination url: " << destination_url;
+  LOG(ERROR) << ": wallpaper id: " << wallpaper_id;
+
+  // TODO(Terry): Ping click confirmation to ads service.
```
Comment (simonhong) [inline components/ntp_background_images/browser/view_counter_service.cc:202]: @tmancey and here.

**Review by tmancey — APPROVED** [review]
LGTM

File: android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java (hunk 1 above)
Comment (simonhong) [reply android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java:624]: @tmancey Changed here. `TabUtils.openUrlInSameTab(mWallpaper.getLogoDestinationUrl());` is added again.

File: components/brave_new_tab_ui/components/default/brandedWallpaper/logo.tsx (hunk 2 above)
Comment (simonhong) [reply components/brave_new_tab_ui/components/default/brandedWallpaper/logo.tsx]: NVM - `href={data.destinationUrl}` change is reverted  and only `onClick` is added.

**Pull request merged**